ecosystem/indexer-grpc/indexer-grpc-fullnode/src/fullnode_data_service.rs

```diff
@@ -61,96 +71,143 @@ impl FullnodeData for FullnodeDataService {
-            let init_status = get_status(StatusType::Init, starting_version, None, ledger_chain_id);
-            match tx.send(Result::<_, Status>::Ok(init_status)).await {
-                Ok(_) => {
-                    // TODO: Add request details later
-                    info!(
-                        start_version = starting_version,
-                        chain_id = ledger_chain_id,
-                        service_type = SERVICE_TYPE,
-                        "[Indexer Fullnode] Init connection"
-                    );
-                },
-                Err(_) => {
-                    panic!("[Indexer Fullnode] Unable to initialize stream");
-                },
-            }
-            let mut base: u64 = 0;
+            let mut current_version = starting_version;
+            let processor_task_count = processor_task_count;
+            let processor_batch_size = processor_batch_size;
+            // Bootstrap the stream.
+            tx.send(Ok(get_status(StatusType::Init, starting_version, None, ledger_chain_id)))
+                .await
+                .unwrap();
+
             loop {
-                let start_time = std::time::Instant::now();
-                // Processes and sends batch of transactions to client
-                let results = coordinator.process_next_batch().await;
-                if results.is_empty() {
-                    info!(
-                        start_version = starting_version,
+                // Nothing for now.
+                let batch_starting_version = current_version;
+                for _ in 0..processor_task_count {
+                    let start_time = std::time::Instant::now();
+                    let mut transactions = vec![CULPRIT_TRANSACTION.clone(); (processor_batch_size as usize)];
+                    // update the versions.
+                    for (i, transaction) in transactions.iter_mut().enumerate() {
+                        transaction.version = current_version + i as u64;
+                    }
+                    // update current version.
+                    current_version += processor_batch_size as u64;
+                    // construct the response.
+                    let response = TransactionsFromNodeResponse {
+                        chain_id: ledger_chain_id as u32,
+                        response: Some(transactions_from_node_response::Response::Data(
+                            TransactionsOutput { transactions },
+                        )),
+                    };
+                    aptos_logger::info!(
+                        start_version = batch_starting_version,
+                        end_version = current_version - 1,
                         chain_id = ledger_chain_id,
-                        "[Indexer Fullnode] Client disconnected."
+                        service_type = SERVICE_TYPE,
+                        duration = start_time.elapsed().as_secs_f64(),
+                        "[Indexer Fullnode] building batch"
```

Comment:
this line is fast

<img width="551" alt="image" src="https://github.com/aptos-labs/aptos-core/assets/112209412/91f6fe69-3cb4-4e2f-b565-3bd083ae0317">